### PR TITLE
refactor: extract db configs and celery configs into dify config

### PR DESCRIPTION
**Pull request by bowenliang123**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

- [x] Tested with db migration tests which relies on db configs and celery configs

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @takatost @JohnJyong

File: api/config.py [hunk 1]
```diff
@@ -3,21 +3,9 @@
...
 DEFAULTS = {
-    'DB_USERNAME': 'postgres',
-    'DB_PASSWORD': '',
-    'DB_HOST': 'localhost',
-    'DB_PORT': '5432',
-    'DB_DATABASE': 'dify',
-    'DB_CHARSET': '',
     'S3_USE_AWS_MANAGED_IAM': 'False',
     'S3_ADDRESS_STYLE': 'auto',
     'SQLALCHEMY_DATABASE_URI_SCHEME': 'postgresql',
```
Comment (takatost) [inline api/config.py]: Perhaps forget to delete this line?

File: api/config.py [hunk 2]
```diff
@@ -69,38 +57,9 @@ def __init__(self):
...
-        self.SQLALCHEMY_DATABASE_URI = f"{self.SQLALCHEMY_DATABASE_URI_SCHEME}://{db_credentials['DB_USERNAME']}:{db_credentials['DB_PASSWORD']}@{db_credentials['DB_HOST']}:{db_credentials['DB_PORT']}/{db_credentials['DB_DATABASE']}{db_extras}"
-        self.SQLALCHEMY_ENGINE_OPTIONS = {
-            'pool_size': int(get_env('SQLALCHEMY_POOL_SIZE')),
-            'max_overflow': int(get_env('SQLALCHEMY_MAX_OVERFLOW')),
-            'pool_recycle': int(get_env('SQLALCHEMY_POOL_RECYCLE')),
-            'pool_pre_ping': get_bool_env('SQLALCHEMY_POOL_PRE_PING'),
-            'connect_args': {'options': '-c timezone=UTC'},
-        }
 
         self.SQLALCHEMY_ECHO = get_bool_env('SQLALCHEMY_ECHO')
```
Comment (takatost) [inline api/config.py]: And this line too.

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/config.py (hunk 1 above)
Comment (bowenliang123) [reply api/config.py]: Nice catch, thanks.

File: api/config.py (hunk 2 above)
Comment (bowenliang123) [reply api/config.py]: Removed.

File: dify.iml
Comment (takatost) [inline dify.iml:1]: These two files below are shown in the wrong place.

**Review by takatost — CHANGES_REQUESTED** [review]

File: dify.iml
Comment (bowenliang123) [reply dify.iml:1]: Already removed. Due to the confusion of peotry settings and IDEA module settings on my local machine.

**Review by takatost — APPROVED** [review]
LGTM

**Pull request merged**